File: NET/LocalApplicationNET/ProducerService/Producer.cs
using System.Text;
using System;
using RabbitMQ.Client;

namespace ProducerService
{
    public class Producer
    {
        public static void SendMessage(string environmnet, string message)
        {
            var factory = new ConnectionFactory() { HostName = environmnet };
            string usedQueue = "Python_Request";
            using (var connection = factory.CreateConnection())
            {
                using (var channel = connection.CreateModel())
                {
                    channel.QueueDeclare(queue: usedQueue,
                                         durable: false,
                                         exclusive: false,
                                         autoDelete: false,
                                         arguments: null);
                    var body = Encoding.UTF8.GetBytes(message);

                    channel.BasicPublish(exchange: "",
                                         routingKey: usedQueue,
                                         basicProperties: null,
                                         body: body);
                    Console.WriteLine("Message Sent", message);
                }
            }
        }
    }
}

File: Python/MessagesLogic/Producer.py
import pika
from MessagesLogic.configurations_support import *


class Producer:
    def __init__(self, message, json_queue="Python", settings_path=r"messagingSettings.json"):
        self.json_RabbitMQ = "RabbitMQ"
        self.json_queues_pointer = "Queues"
        self.json_queue = json_queue + "_Response"

        self.message = message
        self.settings_path = settings_path
        self.prod_state = find_environment_var()
        self.settings_data = read_json_file(self.settings_path)
        self.rabbit_connection_param = self.settings_data[self.json_RabbitMQ][self.prod_state]
        self.used_queue = self.settings_data[self.json_queues_pointer][self.json_queue]

        self.__open_connection()
        self.channel.queue_declare(queue=self.used_queue)
        self.send_message()
        self.__close_connection()

    def __open_connection(self):
        self.connection = pika.BlockingConnection(pika.ConnectionParameters(self.rabbit_connection_param))
        self.channel = self.connection.channel()

    def send_message(self):
        self.channel.basic_publish(exchange="",
                                   routing_key=self.used_queue,
                                   body=self.message,
                                   properties=pika.BasicProperties(delivery_mode=2))  # Makes messages persistent

    def __close_connection(self):
        self.channel.close()
